servo/servo.py

```diff
@@ -435,6 +479,50 @@ def top_level_schema_json(self, *, all: bool = False) -> str:
             default=pydantic.json.pydantic_encoder,
         )
 
+    async def connect(self) -> None:
+        # must define dynamic function to allow configurable backoff
+        await backoff.on_exception(
+            backoff.expo,
+            httpx.HTTPError,
+            max_time=lambda: self.config.settings.backoff.max_time(),
+            max_tries=lambda: self.config.settings.backoff.max_tries(),
+            on_giveup=lambda details: self.logger.critical(
+                f"retries exhausted after {details['elapsed']} seconds, giving up"
+            ),
+        )(self.try_connect)()
+
+    async def try_connect(self) -> None:
+        self.logger.info("Saying HELLO.", end=" ")
+        try:
```

Comment:
what is the end=" " for?
(the log message doesn't seem have a replacement string that looks like '{end}' to be substituted for this kw-arg)

(if this were python's built-in print(), it might have an effect - though not a desirable one).

Reply:
good catch, old protoype cruft https://github.com/opsani/servox/commit/f63f85fc2d6b216f1bb2f5ee38087db5ba958dae#diff-7f314f2efdab903b2e4d8a3ea73e35e671b8a3bd3b344af7b2977b636e99598eL106